.github/workflows/ci.yml

```diff
@@ -1,54 +1,62 @@
+---
 name: "ci"
 
 on:
+  pull_request: {}
   push:
     branches:
-      - "main"
-  pull_request:
-    branches:
-      - "main"
-  workflow_dispatch:
-
-env:
-  CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+      - main
 
 jobs:
-  spec:
+  rubocop_and_matrix:
+    env:
+      BUNDLE_WITHOUT: release
+    runs-on: ubuntu-24.04
+    outputs:
+      ruby: ${{ steps.ruby.outputs.versions }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Ruby ${{ matrix.ruby }}
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "3.4"
+          bundler-cache: true
+      - name: Run Rubocop
+        run: bundle exec rake rubocop
+      - id: ruby
+        uses: voxpupuli/ruby-version@v1
+
+  test:
+    name: "Ruby ${{ matrix.ruby }}"
+    runs-on: ubuntu-24.04
+    needs: rubocop_and_matrix
     strategy:
       fail-fast: false
       matrix:
-        ruby_version:
-          - "3.2"
-        include:
-          - ruby_version: '3.2'
-            puppet_gem_version: '~> 8.0'
-        runs_on:
-          - "ubuntu-latest"
-          - "windows-latest"
-    name: "spec (${{ matrix.runs_on }} ruby ${{ matrix.ruby_version }} | puppet ${{matrix.puppet_gem_version}})"
-    uses: "puppetlabs/cat-github-actions/.github/workflows/gem_ci.yml@main"
-    secrets: "inherit"
-    with:
-      rake_task: "spec:coverage"
-      ruby_version: ${{ matrix.ruby_version }}
-      puppet_gem_version: ${{ matrix.puppet_gem_version }}
-      runs_on: ${{ matrix.runs_on }}
+        ruby: ${{ fromJSON(needs.rubocop_and_matrix.outputs.ruby) }}
+    env:
+      BUNDLE_WITHOUT: release:rubocop
+      BEAKER_HYPERVISOR: docker
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Ruby ${{ matrix.ruby }}
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - name: Build gem
+        run: gem build --strict --verbose *.gemspec
+      - name: Run unit tests
+        run: bundle exec rake spec
+      # acceptance tests currently require litmus
```

Comment:
we can test openvox-strings pretty easily in our modules, I think the acceptance tests don't provide a lot of benefit right now. And they depend on litmus, which depends on bolt, both depend on puppet und a ton of other dependencies.